proof/refine/ARM/Finalise_R.thy

```diff
@@ -2742,7 +2742,9 @@ lemma (in delete_one_conc_pre) finaliseCap_replaceable:
             \<and> (\<forall>p \<in> threadCapRefs cap. st_tcb_at' ((=) Inactive) p s
                      \<and> obj_at' (Not \<circ> tcbQueued) p s
                      \<and> bound_tcb_at' ((=) None) p s
-                     \<and> (\<forall>pr. p \<notin> set (ksReadyQueues s pr))))\<rbrace>"
+                     \<and> (\<forall>pr. p \<notin> set (ksReadyQueues s pr))
+                     \<and> bound_sc_tcb_at' ((=) None or (=) (Some idle_sc_ptr)) p s
+                     \<and> bound_yt_tcb_at' ((=) None) p s))\<rbrace>"
```

Comment:
I think I've seen this condition in a few different lemmas now. It's not relevant to this PR, but we might want to wrap it up in a definition or abbreviation to make it easier to maintain. I haven't fully thought through what it might look like, but it's pretty closely linked to `live'`.